frontend/src/components/Auth/AuthForm.js

```
@@ -0,0 +1,112 @@
+import { useState, useRef, useContext } from "react";
+import { useHistory } from "react-router-dom";
+import AuthContext from "../../store/auth-context";
+
+import classes from "./AuthForm.module.css";
+
+const AuthForm = (props) => {
+  const usernameInputRef = useRef();
+  const passwordInputRef = useRef();
+  const [isLogin, setIsLogin] = useState(true);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const authCtx = useContext(AuthContext);
+  const history = useHistory();
+
+  const switchAuthModeHandler = () => {
+    setIsLogin((prevState) => !prevState);
+  };
+
+  const submitHandler = (event) => {
+    event.preventDefault();
+    const enteredUsername = usernameInputRef.current.value;
+    const enteredPassword = passwordInputRef.current.value;
+    console.log(enteredUsername);
+    console.log(enteredPassword);
+
+    setIsLoading(true);
+    let url;
+    const BACKEND_API = "http://127.0.0.1:4000";
+    if (isLogin) {
+      url = `${BACKEND_API}/login`;
+    } else {
+      url = `${BACKEND_API}/register`;
+    }
+    fetch(url, {
+      method: "POST",
+      body: JSON.stringify({
+        username: enteredUsername,
+        password: enteredPassword,
+        returnSecureToken: true,
+      }),
+      headers: {
+        "Content-Type": "application/json",
+      },
+    })
+      .then((res) => {
+        setIsLoading(false);
+        if (res.ok) {
+          return res.json();
+        } else {
+          return res.json().then(() => {
```

no need to call `res.json()` here. you can just throw an error